Added Strato as Provider

Phiwatec opened the pull request:
I added the german hosting company Strato as a provider.
It is basically the same as google with the url and login params changed.

TimothyYe commented:
Hi @Phiwatec  , I see a new file added `internal/provider/hetzner/hetzner_handler.go` but it's empty. May I know what this file is used for?

Phiwatec commented:
> Hi @Phiwatec , I see a new file added `internal/provider/hetzner/hetzner_handler.go` but it's empty. May I know what this file is used for?

Sorry I messed up that file. Was supposed to be on a different branch. I now deleted the empty file and folder. :)

TimothyYe commented:
> > Hi @Phiwatec , I see a new file added `internal/provider/hetzner/hetzner_handler.go` but it's empty. May I know what this file is used for?
> 
> Sorry I messed up that file. Was supposed to be on a different branch. I now deleted the empty file and folder. :)

I see, thanks.

BTW, to register a new DNS provider, I think you should also update the file `factory.go`, and pay attention to the function  `GetProvider`, all the providers are registered here. So did you test this newly added provider? Does it work well?

Phiwatec commented:
It works now. I updated all necessary files. (Took a bit. Was on vacation :) )
I added one line in google_handler because the system did not handle a wrong password or domain name.
Google and strato return either badauth or nohost when auth failed or the domain to update is unknown respectively.
Now it checks for 'good' or 'nochg' and returns  an error otherwise.

TimothyYe commented on internal/provider/google/google_handler.go:
```diff
@@ -61,13 +61,15 @@ func (provider *DNSProvider) updateIP(domain, subDomain, currentIP string) error
 	if resp.StatusCode != http.StatusOK {
 		log.Errorf("Update IP failed: %s", string(body))
-		return fmt.Errorf("update IP failed: %s", string(body))
+		return fmt.Errorf("Update IP failed: %s", string(body))
```
In Go, error messages are usually written in lower-case.
Please refer to: https://earthly.dev/blog/golang-errors/

TimothyYe commented on internal/provider/google/google_handler.go:
```diff
@@ -61,13 +61,15 @@ func (provider *DNSProvider) updateIP(domain, subDomain, currentIP string) error
 	} else if strings.Contains(string(body), "nochg") {
 		log.Infof("IP not changed: %s", string(body))
+	} else {
+		return fmt.Errorf("Update IP failed: %s", string(body))
```
Same here

TimothyYe commented on internal/provider/strato/strato_handler.go:
```diff
@@ -0,0 +1,76 @@
+	body, _ := io.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		log.Errorf("Update IP failed: %s", string(body))
+		return fmt.Errorf("Update IP failed: %s", string(body))
```
Please use lower-case for the error message.

TimothyYe replied:
Oh, I mean the line 64, not the line 63

Phiwatec replied:
Sorry my bad. Edited the wrong line

TimothyYe commented on internal/provider/strato/strato_handler.go:
```diff
@@ -0,0 +1,76 @@
+	} else if strings.Contains(string(body), "nochg") {
+		log.Infof("IP not changed: %s", string(body))
+	} else {
+		return fmt.Errorf("Update IP failed: %s", string(body))
```
Please use lower-case for the error message.

TimothyYe commented:
@Phiwatec Thank you for your contribution!
